### ISSUE TITLE
Does not handle boolean compares correctly if new value is "false"

### ISSUE THREAD

**Issue by reesz** [netilon/differify#15]
When trying to merge a change in boolean from true to false, differify correctly detects the change, but doesn't apply it correctly.

```js
let differify = new Differify( {
            mode: {
                object: 'DIFF',
                array:  'DIFF',
            },
        } );

const oldObj = { hasDocument: true };
const newObj = { hasDocument: false };

const diff = differify.compare( oldObj, newObj );
console.log( diff );
return differify.applyRightChanges( diff, true );
```

The `compare` function correctly detects the boolean change:

```js
{
  changes: 1,
  status: "MODIFIED",
  _: {
    hasDocument: { original: true, current: false, status: "MODIFIED", changes: 1 }
  }
}
```

But applying the changes with the `applyChanges` function (or any left/right wrapper) doesn't return the new hasDocument prop/value.

I assume this is to line 430 in differify.js:
```js
...
  if (typeof next === 'object') {
    const o = {};
    let curr;
    for (let i in next) {
      if (next.hasOwnProperty(i)) {
        curr = selector(next[i]);
        if (curr) { // <--- If curr is a boolean with the value `false` this if fails and doesn't set `false` as the new value.
          o[i] = curr;
        }
      }
    }

    return o;
  }
...
```

// edit:
Here is my proposed fix for this line, which works currently for me. But I'm not fully into the whole lib, so I don't know if anything else needs to be considered:

```js
      if ( typeof curr !== 'undefined' && curr !== null ) {
```

**Comment by fabian-orue** [comment]
Hi @reesz , thank you for letting me know. I will check it.

**Comment by netilon** [comment]
Hi again @reesz , I have fixed the issue (see the new version 3.2.2). I did something different than your suggestion, because I see that the solution you mentioned wouldn't work with null or undefined properties/values (those are valid options).  Regards.

**Comment by fabian-orue** [comment]
Hi again @reesz , I have fixed the issue (see the new version 3.2.2). I did something different than your suggestion, because I see that the solution you mentioned wouldn't work with null or undefined properties/values (those are valid options).  Regards.
